src/compiler/checker.ts

```diff
@@ -14684,16 +14662,17 @@ namespace ts {
                 if (objectType.flags & (TypeFlags.Any | TypeFlags.Never)) {
                     return objectType;
                 }
-                const stringIndexInfo = getIndexInfoOfType(objectType, IndexKind.String);
-                const indexInfo = isTypeAssignableToKind(indexType, TypeFlags.NumberLike) && getIndexInfoOfType(objectType, IndexKind.Number) || stringIndexInfo;
+                // If no index signature is applicable, we default to the string index signature. In effect, this means the string
+                // index signature applies even when accessing with a symbol-like type.

```

Comment:
😦 Shouldn't we.... fix this?

Reply:
That was my first thought, but no. It only matters for calls to `getIndexedAccessTypeOrUndefined` with no `accessNode` parameter--if we have an `accessNode`, we will indeed report an error when no property or applicable index signature exists. The scenario it enables is having constraints for index accesses involving type parameters constrained by string index signatures. For example, given a type parameter `T extends { [key: string]: boolean }`, we want `T[keyof T]` to have the constraint `boolean`, even though it _technically_ could have symbol-named properties of other types. The constraint is resolved as `T[string | number | symbol]`, which becomes `T[string] | T[number] | T[symbol]`, and `T[symbol]` would spoil the result here without the fallback to the index signature.